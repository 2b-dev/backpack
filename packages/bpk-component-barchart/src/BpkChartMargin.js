import React from 'react';
import PropTypes from 'prop-types';

const BpkChartMargin = (props) => {
  const { children, margin, ...rest } = props;

  return (
    <g
      className="bpk-barchart__margin"
      transform={`translate(${margin.left}, ${margin.top})`}
      {...rest}
    >
      {children}
    </g>
  );
};

BpkChartMargin.propTypes = {
  children: PropTypes.node.isRequired,
  margin: PropTypes.shape({
    top: PropTypes.number,
    bottom: PropTypes.number,
    left: PropTypes.number,
    right: PropTypes.number,
  }).isRequired,
};

export default BpkChartMargin;
