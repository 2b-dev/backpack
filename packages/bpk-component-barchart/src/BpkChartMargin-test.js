import React from 'react';
import renderer from 'react-test-renderer';
import BpkChartMargin from './BpkChartMargin';

const margin = {
  top: 10,
  right: 20,
  bottom: 30,
  left: 40,
};

describe('BpkChartMargin', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkChartMargin
        margin={margin}
      >Children</BpkChartMargin>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
