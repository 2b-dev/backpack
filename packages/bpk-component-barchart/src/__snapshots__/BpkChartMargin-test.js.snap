exports[`BpkChartMargin should render correctly 1`] = `
<g
  className="bpk-barchart__margin"
  transform="translate(40, 10)">
  Children
</g>
`;
